### ISSUE TITLE
多个sample结果不同

### ISSUE THREAD

**Issue by Le-Code** [Morizeyao/GPT2-Chinese#115]
为什么可以生成多个不同的sample？ 调试时发现，每个sample下模型的输入都是一样的但是输出结果却不同，这是因为什么呢？

**Comment by echoyes** [comment]
> 为什么可以生成多个不同的sample？ 调试时发现，每个sample下模型的输入都是一样的但是输出结果却不同，这是因为什么呢？

那是因为用了torch.multinomial这个采样函数吧，你换成torch.tensor([torch.argmax(F.softmax(filtered_logits, dim=-1))]).to(device) 试试

**Comment by Morizeyao** [comment]
随机取样是会随机生成的
